File: app/src/test/java/com/annushkaproject/programmerscalculator/ProgrammerTests/ProgrammerMultiplicationTest.java
package com.annushkaproject.programmerscalculator.ProgrammerTests;

import com.annushkaproject.programmerscalculator.model.ProgrammerCalcModel;
import com.annushkaproject.programmerscalculator.utils.ProgrammerOperationsUtil;

import org.junit.Test;

import java.math.BigDecimal;

import static com.annushkaproject.programmerscalculator.model.Operator.MULTIPLY;
import static com.annushkaproject.programmerscalculator.model.int_size_enum.*;
import static org.junit.Assert.*;

public class ProgrammerMultiplicationTest {
    @Test
    public void programmerMultiplicationQWORD_isCorrect() {
        assertEquals(25, ProgrammerOperationsUtil.calculateWithData(new ProgrammerCalcModel(new BigDecimal(5), new BigDecimal(5), MULTIPLY, len8)));
        assertEquals(-25, ProgrammerOperationsUtil.calculateWithData(new ProgrammerCalcModel(new BigDecimal(-5), new BigDecimal(5), MULTIPLY, len8)));
        assertEquals(0x8000000000000000L, ProgrammerOperationsUtil.calculateWithData(new ProgrammerCalcModel(new BigDecimal(0x4000000000000000L), new BigDecimal(2), MULTIPLY, len8)));
        assertEquals(0, ProgrammerOperationsUtil.calculateWithData(new ProgrammerCalcModel(new BigDecimal(0x8000000000000000L), new BigDecimal(0x8000000000000000L), MULTIPLY, len8)));
        assertEquals(0xFFFFFFFFFFFFFFFEL, ProgrammerOperationsUtil.calculateWithData(new ProgrammerCalcModel(new BigDecimal(0x7FFFFFFFFFFFFFFFL), new BigDecimal(2), MULTIPLY, len8)));
        assertEquals(0x7FFFFFFFFFFFFFFEL, ProgrammerOperationsUtil.calculateWithData(new ProgrammerCalcModel(new BigDecimal(0x3FFFFFFFFFFFFFFFL), new BigDecimal(2), MULTIPLY, len8)));
    }

    @Test
    public void programmerMultiplicationDWORD_isCorrect() {
        assertEquals(25, ProgrammerOperationsUtil.calculateWithData(new ProgrammerCalcModel(new BigDecimal(5), new BigDecimal(5), MULTIPLY, len4)));
        assertEquals(-25, ProgrammerOperationsUtil.calculateWithData(new ProgrammerCalcModel(new BigDecimal(-5), new BigDecimal(5), MULTIPLY, len4)));
        assertEquals(0x80000000, ProgrammerOperationsUtil.calculateWithData(new ProgrammerCalcModel(new BigDecimal(0x40000000), new BigDecimal(2), MULTIPLY, len4)));
        assertEquals(0, ProgrammerOperationsUtil.calculateWithData(new ProgrammerCalcModel(new BigDecimal(0x80000000), new BigDecimal(0x80000000), MULTIPLY, len4)));
        assertEquals(0xFFFFFFFE, ProgrammerOperationsUtil.calculateWithData(new ProgrammerCalcModel(new BigDecimal(0x7FFFFFFF), new BigDecimal(2), MULTIPLY, len4)));
        assertEquals(0x7FFFFFFE, ProgrammerOperationsUtil.calculateWithData(new ProgrammerCalcModel(new BigDecimal(0x3FFFFFFF), new BigDecimal(2), MULTIPLY, len4)));
    }

    @Test
    public void programmerMultiplicationWORD_isCorrect() {
        assertEquals((short) 25, ProgrammerOperationsUtil.calculateWithData(new ProgrammerCalcModel(new BigDecimal(5), new BigDecimal(5), MULTIPLY, len2)));
        assertEquals((short) -25, ProgrammerOperationsUtil.calculateWithData(new ProgrammerCalcModel(new BigDecimal(-5), new BigDecimal(5), MULTIPLY, len2)));
        assertEquals((short) 0x8000, ProgrammerOperationsUtil.calculateWithData(new ProgrammerCalcModel(new BigDecimal(0x4000), new BigDecimal(2), MULTIPLY, len2)));
        assertEquals((short) 0, ProgrammerOperationsUtil.calculateWithData(new ProgrammerCalcModel(new BigDecimal(0x8000), new BigDecimal(0x8000), MULTIPLY, len2)));
        assertEquals((short) 0xFFFE, ProgrammerOperationsUtil.calculateWithData(new ProgrammerCalcModel(new BigDecimal(0x7FFF), new BigDecimal(2), MULTIPLY, len2)));
        assertEquals((short) 0x7FFE, ProgrammerOperationsUtil.calculateWithData(new ProgrammerCalcModel(new BigDecimal(0x3FFF), new BigDecimal(2), MULTIPLY, len2)));
    }

    @Test
    public void programmerMultiplicationBYTE_isCorrect() {
        assertEquals((byte) 25, ProgrammerOperationsUtil.calculateWithData(new ProgrammerCalcModel(new BigDecimal(5), new BigDecimal(5), MULTIPLY, len1)));
        assertEquals((byte) -25, ProgrammerOperationsUtil.calculateWithData(new ProgrammerCalcModel(new BigDecimal(-5), new BigDecimal(5), MULTIPLY, len1)));
        assertEquals((byte) 0x80, ProgrammerOperationsUtil.calculateWithData(new ProgrammerCalcModel(new BigDecimal(0x40), new BigDecimal(2), MULTIPLY, len1)));
        assertEquals((byte) 0, ProgrammerOperationsUtil.calculateWithData(new ProgrammerCalcModel(new BigDecimal(0x80), new BigDecimal(0x80), MULTIPLY, len1)));
        assertEquals((byte) 0xFE, ProgrammerOperationsUtil.calculateWithData(new ProgrammerCalcModel(new BigDecimal(0x7F), new BigDecimal(2), MULTIPLY, len1)));
        assertEquals((byte) 0x7E, ProgrammerOperationsUtil.calculateWithData(new ProgrammerCalcModel(new BigDecimal(0x3F), new BigDecimal(2), MULTIPLY, len1)));
    }
}

File: app/src/test/java/com/annushkaproject/programmerscalculator/ProgrammerTests/ProgrammerMultiplicationTest.kt
package com.annushkaproject.programmerscalculator.ProgrammerTests

import com.annushkaproject.programmerscalculator.model.*
import com.annushkaproject.programmerscalculator.utils.ProgrammerOperationsUtil.calculateWithData
import org.junit.Assert
import org.junit.Test
import java.math.BigDecimal

class ProgrammerMultiplicationTest {
    @Test
    fun programmerMultiplicationQWORD_isCorrect() {
        Assert.assertEquals(25, calculateWithData(ProgrammerCalcModel(BigDecimal(5), BigDecimal(5), Operator.MULTIPLY, int_size_enum.l8)))
        Assert.assertEquals(-25, calculateWithData(ProgrammerCalcModel(BigDecimal(-5), BigDecimal(5), Operator.MULTIPLY, int_size_enum.l8)))
        Assert.assertEquals(-0x8000000000000000L, calculateWithData(ProgrammerCalcModel(BigDecimal(0x4000000000000000L), BigDecimal(2), Operator.MULTIPLY, int_size_enum.l8)))
        Assert.assertEquals(0, calculateWithData(ProgrammerCalcModel(BigDecimal(-0x8000000000000000L), BigDecimal(-0x8000000000000000L), Operator.MULTIPLY, int_size_enum.l8)))
        Assert.assertEquals(-0x2L, calculateWithData(ProgrammerCalcModel(BigDecimal(0x7FFFFFFFFFFFFFFFL), BigDecimal(2), Operator.MULTIPLY, int_size_enum.l8)))
        Assert.assertEquals(0x7FFFFFFFFFFFFFFEL, calculateWithData(ProgrammerCalcModel(BigDecimal(0x3FFFFFFFFFFFFFFFL), BigDecimal(2), Operator.MULTIPLY, int_size_enum.l8)))
    }

    @Test
    fun programmerMultiplicationDWORD_isCorrect() {
        Assert.assertEquals(25, calculateWithData(ProgrammerCalcModel(BigDecimal(5), BigDecimal(5), Operator.MULTIPLY, int_size_enum.l4)))
        Assert.assertEquals(-25, calculateWithData(ProgrammerCalcModel(BigDecimal(-5), BigDecimal(5), Operator.MULTIPLY, int_size_enum.l4)))
        Assert.assertEquals(-0x80000000, calculateWithData(ProgrammerCalcModel(BigDecimal(0x40000000), BigDecimal(2), Operator.MULTIPLY, int_size_enum.l4)))
        Assert.assertEquals(0, calculateWithData(ProgrammerCalcModel(BigDecimal(-0x80000000), BigDecimal(-0x80000000), Operator.MULTIPLY, int_size_enum.l4)))
        Assert.assertEquals(-0x2, calculateWithData(ProgrammerCalcModel(BigDecimal(0x7FFFFFFF), BigDecimal(2), Operator.MULTIPLY, int_size_enum.l4)))
        Assert.assertEquals(0x7FFFFFFE, calculateWithData(ProgrammerCalcModel(BigDecimal(0x3FFFFFFF), BigDecimal(2), Operator.MULTIPLY, int_size_enum.l4)))
    }

    @Test
    fun programmerMultiplicationWORD_isCorrect() {
        Assert.assertEquals(25 as Short.toLong(), calculateWithData(ProgrammerCalcModel(BigDecimal(5), BigDecimal(5), Operator.MULTIPLY, int_size_enum.l2)))
        Assert.assertEquals(-25 as Short.toLong(), calculateWithData(ProgrammerCalcModel(BigDecimal(-5), BigDecimal(5), Operator.MULTIPLY, int_size_enum.l2)))
        Assert.assertEquals(0x8000 as Short.toLong(), calculateWithData(ProgrammerCalcModel(BigDecimal(0x4000), BigDecimal(2), Operator.MULTIPLY, int_size_enum.l2)))
        Assert.assertEquals(0 as Short.toLong(), calculateWithData(ProgrammerCalcModel(BigDecimal(0x8000), BigDecimal(0x8000), Operator.MULTIPLY, int_size_enum.l2)))
        Assert.assertEquals(0xFFFE as Short.toLong(), calculateWithData(ProgrammerCalcModel(BigDecimal(0x7FFF), BigDecimal(2), Operator.MULTIPLY, int_size_enum.l2)))
        Assert.assertEquals(0x7FFE as Short.toLong(), calculateWithData(ProgrammerCalcModel(BigDecimal(0x3FFF), BigDecimal(2), Operator.MULTIPLY, int_size_enum.l2)))
    }

    @Test
    fun programmerMultiplicationBYTE_isCorrect() {
        Assert.assertEquals(25 as Byte.toLong(), calculateWithData(ProgrammerCalcModel(BigDecimal(5), BigDecimal(5), Operator.MULTIPLY, int_size_enum.l1)))
        Assert.assertEquals(-25 as Byte.toLong(), calculateWithData(ProgrammerCalcModel(BigDecimal(-5), BigDecimal(5), Operator.MULTIPLY, int_size_enum.l1)))
        Assert.assertEquals(0x80 as Byte.toLong(), calculateWithData(ProgrammerCalcModel(BigDecimal(0x40), BigDecimal(2), Operator.MULTIPLY, int_size_enum.l1)))
        Assert.assertEquals(0 as Byte.toLong(), calculateWithData(ProgrammerCalcModel(BigDecimal(0x80), BigDecimal(0x80), Operator.MULTIPLY, int_size_enum.l1)))
        Assert.assertEquals(0xFE as Byte.toLong(), calculateWithData(ProgrammerCalcModel(BigDecimal(0x7F), BigDecimal(2), Operator.MULTIPLY, int_size_enum.l1)))
        Assert.assertEquals(0x7E as Byte.toLong(), calculateWithData(ProgrammerCalcModel(BigDecimal(0x3F), BigDecimal(2), Operator.MULTIPLY, int_size_enum.l1)))
    }
}